### PR TITLE
Create markdown viewer

### PR DESCRIPTION
**Pull request by Kiarokh**
## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/markdown-viewer/markdown-viewer.tsx
Comment (Kiarokh) [inline src/components/markdown-viewer/markdown-viewer.tsx:21]:
```suggestion
    tag: 'limel-markdown',
```

Comment (Kiarokh) [reply src/components/markdown-viewer/markdown-viewer.tsx:21]: Let's rename all files and tag names.

File: src/components/markdown/examples/markdown-headings.tsx
Comment (Kiarokh) [inline src/components/markdown/examples/markdown-headings.tsx:17]: these aren't working, so I'll remove them

File: src/components/markdown/examples/markdown-blockquotes.tsx
Comment (Kiarokh) [inline src/components/markdown/examples/markdown-blockquotes.tsx:1]: source code for the examples are taken from here: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet 

to verify that everything renders fine.

File: src/components/markdown/examples/markdown-footnotes.tsx
Comment (Kiarokh) [inline src/components/markdown/examples/markdown-footnotes.tsx:15]: according to the source, they should be rendered like this 
image
But not sure how to test and see them. 🤔 maybe we should just remove this example.

File: src/components/markdown/examples/markdown-html.tsx
Comment (Kiarokh) [inline src/components/markdown/examples/markdown-html.tsx:1]: ain't rendering currently

File: src/components/markdown/markdown.tsx
Comment (jgroth) [inline src/components/markdown/markdown.tsx:31]: There are both pros and cons for having this as a prop vs having the content provided as child elements instead. How did the discussion go about this topic when it was implemented?

Pros for having it as child elements is that it makes more sense, especially if HTML is allowed as well. Text is flowing content, and forcing the consumer to put it all in a string first is cumbersome. E.g.

```html
<limel-markdown>
    Hello, 
    <span class="green">World!</span>
</limel-markdown>
```
vs
```html
<limel-markdown 
    content="Hello, \n<span class=\"green\">World!</span>">
</limel-markdown>
```

Cons might be that it is a bit harder to implement for us.

If we insist on keeping this as a prop, I think we should make it more compatible with our other components and name this `value` instead of `content`

Comment (Kiarokh) [reply src/components/markdown/markdown.tsx:31]: > If we insist on keeping this as a prop, I think we should make it more compatible with our other components and name this value instead of content

Agree. Check this out, kind of relevant https://github.com/Lundalogik/crm-feature/issues/3480#issuecomment-1724922053

Comment (john-traas) [reply src/components/markdown/markdown.tsx:31]: If I recall correctly, we went with passing it as a prop because it allowed us to escape the linter/formatter that would automatically reformat the markdown text, which would then break the markdown. 
The only way to force the linter not to automatically reformat everything was a succession of ugly 'ignore' comments.

Because of that, we felt made more sense to pass it in as a prop.

Comment (jgroth) [reply src/components/markdown/markdown.tsx:31]: I still think we should rename the `content` prop to `value`

Comment (Kiarokh) [reply src/components/markdown/markdown.tsx:31]: will u do this @john-traas please? there seems to be some conflicts in the branch as well, that needs resolving.

Comment (john-traas) [reply src/components/markdown/markdown.tsx:31]: Yes, I am on it @Kiarokh

File: src/components/markdown/examples/markdown-footnotes.tsx
Comment (john-traas) [reply src/components/markdown/examples/markdown-footnotes.tsx:15]: I have got footnotes working @Kiarokh 🥳

Comment (Kiarokh) [reply src/components/markdown/examples/markdown-footnotes.tsx:15]: Wonderful! <3

File: src/components/markdown/markdown.tsx
Comment (adrianschmidt) [reply src/components/markdown/markdown.tsx]: It looks to me like this was resolved in favour of keeping it as a property, rather than a slot, but just in case:

I would assume that anytime someone wants to use this element, they have the content they need to present as a string, most likely from the database?

Also, I don't see why you'd ever need to put html into this component if you already have it as a DOM-structure? If it's for the sake of sanitising the html, that will already be too late by the time this element picks it up, as you've already injected the html into the DOM before the slot can "pick it up" for processing. (Or at least I'd presume it would be too late? And even if it wasn't, we'd presumably only be a small change to the `slot` functionality away from a security problem.)

Anyway, this might feel like I'm trying to hammer home that I think using `slot` is a bad idea, but it's more that I'm wondering if there's something I'm missing? 🤗

File: package-lock.json
Comment (adrianschmidt) [inline package-lock.json:1]: @john-traas I've co-assigned myself to this PR, as I'd like to be involved before we merge it. This is a component we've wanted for a long time. I think it will start to be used pretty much immediately when we add it, and it also has security implications, so getting it right feels pretty important 😊

Comment (adrianschmidt) [reply package-lock.json:1]: (Just based on the packages added though, this looks promising 😊 We seem to have used the same things we already used for the history note stuff 👍)

Comment (john-traas) [reply package-lock.json:1]: @adrianschmidt that's great, we definitely need to get it right. 😃

File: src/components/markdown/markdown.tsx
Comment (john-traas) [reply src/components/markdown/markdown.tsx]: I think we're on the same page, keeping it as a `property` is the approach we're going with.

File: package-lock.json
Comment (adrianschmidt) [reply package-lock.json:1]: Sorry, I don't think I'll have the time to look at this one in any depth. It looks like we've basically lifted the implementation from the one we currently use for history in the web client, and if so, I guess this should be fine 👍

**Comment by Kiarokh** [comment]
So does it mean you have approved this now @adrianschmidt ? :)

File: stencil.config.ts
Comment (jgroth) [inline stencil.config.ts:17]: Did we still need this? If so, should it not be needed in the other config file as well? :slightly_smiling_face:

Comment (john-traas) [reply stencil.config.ts]: We don't need it anymore so I removed it.

**Review by jgroth — APPROVED** [review]

**Review by john-traas — APPROVED** [review]

File: src/components/markdown/examples/markdown-code.tsx
Comment (adrianschmidt) [reply src/components/markdown/examples/markdown-code.tsx:3]: Doesn't that work with single backticks?

Like this:
```js
const markdown = 'testing some `code` here';
```

Or like this:
```js
const markdown = `
    testing some \`code\` here
`;
```

**Comment by adrianschmidt** [comment]
> So does it mean you have approved this now @adrianschmidt ? :)

I can't approve it since I haven't reviewed it, but I certainly don't _disapprove_! 😄

**Pull request merged**
